### PR TITLE
CCEE - HAP for Openstack with region

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**

For `openstack` provider we need separate subscription pool for every region.

Changes proposed in this pull request:

- hyperscalerType with optional region component
- in sharedGardenerPool for `openstack` provider we fetch subscriptions using key/label of form `openstack_<region>`

**Related issue(s)**

### CODE REVIEW

File: common/hyperscaler/hyperscalerType.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	}
+}
+
+func AWS() Type {
+	return Type{
+		hyperscalerName: "aws",
+	}
+}
+
+func Openstack() Type {
```
Comment (piotrmiskiewicz) [inline common/hyperscaler/hyperscalerType.go:26]: func Openstack(region string) Type {

**Comment by jaroslaw-pieszka** [comment]
test pre-kyma-environment-broker

File: cmd/broker/suite_test.go [hunk 2]
```diff
@@ -1029,20 +1030,80 @@ func fixConfig() *Config {
 		Update:         process.StagedManagerConfiguration{MaxStepProcessingTime: time.Minute},
 	}
 }
+func fixBrokerConfigWithOptionalRegion() *Config {
```
Comment (piotrmiskiewicz) [inline cmd/broker/suite_test.go:1033]: please, use fixConfig and modify it if possible

Comment (jaroslaw-pieszka) [reply cmd/broker/suite_test.go]: Fixed in the next commit.

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**